pandas/core/groupby/generic.py

```diff
@@ -856,23 +856,28 @@ def _get_index():
             return index
 
         if isinstance(values[0], dict):
-            # GH #823
+            # GH #823 #24880
             index = _get_index()
-            result = DataFrame(values, index=index).stack()
+            result = self._reindex_output(DataFrame(values, index=index))
+            # if self.observed is False,
+            # keep all-NaN rows created while re-indexing
+            result = result.stack(dropna=self.observed)
             result.name = self._selection_name
             return result
 
-        if isinstance(values[0], (Series, dict)):
+        if isinstance(values[0], Series):
```

Comment:
Why was dict removed here?

Reply:
There is a `if isinstance(values[0], dict):` just above that, so I think it will never reach the second condition if it's a dict. I think it's redundant in this case.

Reply:
Gotcha - good find